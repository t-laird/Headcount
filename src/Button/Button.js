import React from 'react';
import PropTypes from 'prop-types';


const Button = (props) => {
  return (
    <button 
      className={props.type}
      onClick={() => {
        if (!props.chartStatus) {
          props.changeData(props.type, props.data);
        } else {
          props.chartStatus(false);
        }
      }}>
      {props.name}
    </button>
  )
}

Button.propTypes = {
  data: PropTypes.array,
  type: PropTypes.string,
  name: PropTypes.string,
  populateData: PropTypes.func,
  displayDataLabel: PropTypes.func
};

export default Button;