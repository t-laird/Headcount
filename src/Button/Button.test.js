/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import { shallow, mount } from 'enzyme';

describe('Button test', () => {
  it('should render the correctly in Nav', () => {
    const mockFunc = jest.fn();

    const renderedButton = shallow(
      <Button
        icon='icon-child'
        type='kinderData' 
        buttonClass='kinderData' 
        changeData={mockFunc}
        name='Kindergartens in full day programs'
        data={[]} />); 

    expect(renderedButton.find('.kinderData').length).toEqual(1);
    expect(renderedButton.find('i').length).toEqual(1);
  });

  it('should render correctly in Chart', () => {
    const mockFunc = jest.fn();

    const renderedButton = shallow(
      <Button 
        buttonClass='chart-close'
        type='chart-close'
        chartStatus={mockFunc}
        name='Close' />);

    expect(renderedButton.find('.chart-close').length).toEqual(1);
  });

  it('should match the Button snapshot', () => {
    const mockFunc = jest.fn();

    const renderedButton = shallow(
      <Button 
        icon='icon-child'
        type='kinderData' 
        changeData={mockFunc}
        name='Kindergartens in full day programs'
        data={[]}/>);

    expect(renderedButton).toMatchSnapshot();
  });

  it('Should display the correct label based on props in Nav', () => {
    const mockFunc = jest.fn();

    const renderedButton = shallow(
      <Button 
        icon='icon-child'
        type='kinderData' 
        changeData={mockFunc}
        name='Kindergarteners in full day programs'
        data={[]}/>);
    expect(renderedButton.text()).toEqual('Kindergarteners in full day programs')
  });

  it('Should display the correct label based on props in Chart', () => {
    const mockFunc = jest.fn();

    const renderedButton = shallow(
      <Button 
        type='chart-close'
        chartStatus={mockFunc}
        name='Close' />);

    expect(renderedButton.text()).toEqual('Close')
  });

  it('should call the correct method passed from props in the onClick when inside Nav', () => {
    const mockFunc = jest.fn();

    const mockFunc2 = jest.fn();

    const renderedButton = shallow(
      <Button 
        icon='icon-child'
        type='kinderData' 
        changeData={mockFunc}
        name='Kindergartens in full day programs'
        data={[]}/>);

    renderedButton.simulate('click');
    expect(mockFunc.mock.calls.length).toEqual(1);
    expect(mockFunc2.mock.calls.length).toEqual(0);
  });

  it('should call the correct method passed from props in the onClick when inside Chart', () => {
    const mockFunc = jest.fn();

    const mockFunc2 = jest.fn();

    const renderedButton = shallow(
      <Button 
        type='chart-close'
        chartStatus={mockFunc2}
        name='Close' />);

    renderedButton.simulate('click');
    expect(mockFunc.mock.calls.length).toEqual(0);
    expect(mockFunc2.mock.calls.length).toEqual(1);
  });
});
