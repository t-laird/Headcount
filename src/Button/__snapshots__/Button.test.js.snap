exports[`Button test should match the Button snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    changeData={[Function]}
    data={Array []}
    icon="icon-child"
    name="Kindergartens in full day programs"
    type="kinderData" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          className="icon-child" />,
        "Kindergartens in full day programs",
      ],
      "className": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon-child",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "Kindergartens in full day programs",
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="icon-child" />,
          "Kindergartens in full day programs",
        ],
        "className": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon-child",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "Kindergartens in full day programs",
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
